app/Console/Commands/SupportEvents.php

```diff
@@ -0,0 +1,197 @@
+
+        foreach ($this->event_pull_lut["all_events"] as $artcc => $airports) {
+            foreach ($airports as $airport) {
+                $all_events_lut[] = $airport;
+                $inverse_lut[$airport] = $artcc;
+            }
+        }
+        foreach ($this->event_pull_lut["fno_or_live_only"] as $artcc => $airports) {
+            foreach ($airports as $airport) {
+                $fno_or_live_only_lut[] = $airport;
+                $inverse_lut[$airport] = $artcc;
+            }
+        }
+
+
+        $client = new Client();
+
+        $this->info('Downloading event info from VATSIM');
+
+        $res = $client->get('https://my.vatsim.net/api/v1/events/all');
+
+        $this->info('Parsing event info JSON');
+
+        $result = json_decode($res->getBody());
+        if (is_null($result)) {
+            return;
+        }
+
+        foreach ($result->data as $event) {
+            $existing = Event::where('vatsim_id', $event->id)->first();
+            if ($existing !== null) {
+                $this->info("Skipping already processed event ".$event->id);
+                continue;
+            } // this event has already been processed
+
+            $pull_this_event = false;
+            $organizer = null;
+
+            // parse times
+            $start_time = Carbon::parse($event->start_time);
+            $end_time = Carbon::parse($event->end_time);
+
+            foreach ($event->airports as $airport) {
+                if (in_array($airport->icao, $all_events_lut)) {
+                    $pull_this_event = true;
+                    $organizer = $inverse_lut[$airport->icao];
+                    break;
+                }
+                if (in_array($airport->icao, $fno_or_live_only_lut)) {
+                    // is it a live event?
+                    if (str_contains(strtolower($event->name), 'live')) {
+                        // live event - pull this event too
+                        $pull_this_event = true;
+                        $organizer = $inverse_lut[$airport->icao];
+                        break;
+                    }
+                    // is it on a friday? (FNO)
+                    if ($start_time->dayOfWeek == 6) { // 06: Friday
+                        // FNO - pull this event too
+                        $pull_this_event = true;
+                        $organizer = $inverse_lut[$airport->icao];
+                        break;
+                    }
+                }
+            }
+
+            if (!$pull_this_event) {
+                continue;
+            }
+
+            $this->info('Creating support event with vatsim id '.$event->id);
+
+            // download the event banner
+            // public/storage/event_banners/vatsim_ID.ext
+
+            $this->info($event->id.': Downloading banner');
+
+            $public_url = '/event_banners/vatsim_'.$event->id.substr($event->banner, -4);
+
+            Storage::disk('public')->put($public_url, file_get_contents($event->banner));
```

Comment:
I would recommend maybe storing the VATSIM banner URL here so we don't cram our storage full of other banners

Reply:
Done!